tests/acceptance/features/bootstrap/Provisioning.php

```diff
@@ -364,46 +366,114 @@ public function theseUsersHaveBeenCreatedWithoutSkeletonFiles(TableNode $table)
+			$this->ldap->add($items['dn'], $items);
+		} else {
+			foreach ($items as $item) {
+				$this->ldap->add($item['dn'], $item);
+			}
+		}
+	}
+
+	/**
+	 * @param $suiteParameters
+	 *
 	 * @throws \Exception
+	 * @throws \LdapException
 	 */
-	public function theseUsersHaveBeenCreated($setDefaultAttributes, $doNotInitialize, TableNode $table) {
-		$this->verifyTableNodeColumns($table, ['username'], ['displayname', 'email', 'password']);
-		$table = $table->getColumnsHash();
-		$setDefaultAttributes = $setDefaultAttributes !== "";
-		$initialize = $doNotInitialize === "";
-		// We add all the request bodies in an array.
-		$bodies = [];
-		// We add all the request objects in an array so that we can send all the requests in parallel.
-		$requests = [];
-		$client = new Client();
+	public function connectToLdap($suiteParameters) {
+		SetupHelper::init(
+			$this->getAdminUsername(),
+			$this->getAdminPassword(),
+			$this->getBaseUrl(),
+			$this->getOcPath()
+		);
+		$occResult = SetupHelper::runOcc(
+			['ldap:show-config', 'LDAPTestId', '--output=json']
+		);
+		Assert::assertSame(
+			'0', $occResult['code'],
+			"could not read current LDAP config. stdOut: " .
+			$occResult['stdOut'] .
+			" stdErr: " . $occResult['stdErr']
+		);
 
-		if (\getenv("TEST_EXTERNAL_USER_BACKENDS") === "true") {
-			$users = [];
-			echo "creating LDAP users is not implemented, so assume they exist\n";
-			foreach ($table as $user) {
-				\array_push($users, $user["username"]);
-			}
-			$this->initializeUserBatch($users);
-			return;
+		$ldapConfig = \json_decode(
+			$occResult['stdOut'], true
+		);
+		Assert::assertNotNull(
+			$ldapConfig,
+			"could not json decode current LDAP config. stdOut: " . $occResult['stdOut']
+		);
+
+		$this->ldapBaseDN = (string)$ldapConfig['ldapBase'][0];
+		$this->ldapHost = (string)$ldapConfig['ldapHost'];
+		$this->ldapPort = (string)$ldapConfig['ldapPort'];
+		$this->ldapAdminUser = (string)$ldapConfig['ldapAgentName'];
+
+		$this->ldapAdminPassword = (string)$suiteParameters['ldapAdminPassword'];
+		$this->ldapUsersOU = (string)$suiteParameters['ldapUsersOU'];
+		$this->ldapGroupsOU = (string)$suiteParameters['ldapGroupsOU'];
+
+		$options = [
+			'host' => $this->ldapHost,
+			'port' => $this->ldapPort,
+			'password' => $this->ldapAdminPassword,
+			'bindRequiresDn' => true,
+			'baseDn' => $this->ldapBaseDN,
+			'username' => $this->ldapAdminUser
+		];
+		$this->ldap = new Ldap($options);
+		$this->ldap->bind();
+
+		$this->importLdifFile(
+			__DIR__ . (string)$suiteParameters['ldapInitialUserFilePath']
+		);
+		$this->theLdapUsersHaveBeenResynced();
+	}
+
+	/**
+	 * @throws Exception
+	 */
+	public function theLdapUsersHaveBeenReSynced() {
+		$occResult = SetupHelper::runOcc(
+			['user:sync', 'OCA\User_LDAP\User_Proxy', '-m', 'remove']
+		);
+		if ($occResult['code'] !== "0") {
+			throw new \Exception("could not sync LDAP users " . $occResult['stdErr']);
 		}
+	}
 
+	/**
+	 * @param $setDefaultAttributes
```

Comment:
Its better to use parameter type as well here. eg.
```suggestion
	 * @param boolean $setDefaultAttributes
```

Reply:
done!